### ISSUE TITLE
Issues when used compression-webpack-plugin

### ISSUE THREAD

**Issue by webdiscus** [webdiscus/html-bundler-webpack-plugin#134]
Using [compression-webpack-plugin](https://github.com/webpack-contrib/compression-webpack-plugin) will be generate a `*.html.gz` file where:

## 1) HTML content is not minified although the `minify` option is enabled.

The HTML file should be minified.

## 2) Wrong output script URL

The source JS file will be resolved and replaced with absolute path instead of public output URL

There is a template:
```html
<script src="@src/js/app.js"></script>
```
Expected:
```html
<script src="js/app.83b54efd8.js"></script>
```

Received:
```html
<script src="/project/src/js/app.js"></script>
```

Both issues are only in `*.html.gz` file, the `*.html` generated file is ok.

---
Here is the [repository](https://github.com/vralle/webpack-html-bundler-starting-template/tree/issues), the branch is  `issues` .

**Comment by webdiscus** [comment]
Hello @vralle,

I know why the problem occurs.
The problem is in the order of the hooks being called.
I'll try to analyze if it's possible to fix it in the bundler plugin.

**Comment by webdiscus** [comment]
@vralle,

I can fix it in the plugin using an earlier rendering stage than `PROCESS_ASSETS_STAGE_OPTIMIZE_TRANSFER`, e.g. `PROCESS_ASSETS_STAGE_OPTIMIZE_TRANSFER - 1`.

I will add an option to set the rendering stage for such specific situations.

**Comment by webdiscus** [comment]
@vralle,

the issue for this special use case is fixed in `v4.11.0`, added the new [renderStage](https://github.com/webdiscus/html-bundler-webpack-plugin/issues/134#option-renderStage) option.
The  compression-webpack-plugin works this the default value of this  option.

Can you please test it in your project?

**Comment by vralle** [comment]
@webdiscus, with the plugin v4.11.0 the compilation result corresponds to expectations.

In the plugin configuration, typescript reports that `renderStage` declaration is required.

**Comment by webdiscus** [comment]
@vralle 

in TypeScript the `renderStage` option should be optional, fixed in `v4.11.1`
